File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer> hm = new HashSet<Integer>();
        for (int num: nums){
            if (hm.contains(num)) return true;
            hm.add(num);
        }
        return nums.length != hm.size();
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> uSet(nums.begin(), nums.end());
        int res = 0;
        for (int num: uSet){
            if (!uSet.contains(num-1)){
                int currNum = num;
                int length = 1;

                while (uSet.contains(currNum+1)){
                    currNum++;
                    length++;
                }

                res = max(res, length);
            }
        }

        return res;
    }
};
